lib/Nexus.js

```diff
@@ -268,6 +269,42 @@ module.exports = function xGraph(__options = {}) {
 				return pid;
 			}
 
+			/**
+			 * Scan the array of Pids and send a 'GetServices'
+			 * command to each in turn to accumulate the 
+			 * services used by a partifcular module
+			 * @param {array} services		array of Pids to send GetServices to
+			 */
+			async function getServices(services) {
+				let svc = {};
+				let com = {};
+				
+				return new Promise(async (resolve, _reject) => {
+					for (let is = 0; is < services.length; is++) {
+						let pid = services[is];
+						await service(pid);
+					}
+					resolve(svc);
+				});
+
```

Comment:
please add some documentation here as to what this function does. it is unclear, and not easily implied from the code. The function takes a pid and returns nothing. we can safely assume its getting the services from the pid in the parameters, but the function returns a promise, that returns a promise, that returns nothing, always. so its functionality is quite obfuscated.